File: frameNet/src/main/java/org/sqlunet/framenet/FnPatternPointer.java
package org.sqlunet.framenet;

import android.os.Parcel;
import android.os.Parcelable;
import android.support.annotation.NonNull;

import org.sqlunet.Pointer;

/**
 * Parcelable pattern
 *
 * @author <a href="mailto:1313ou@gmail.com">Bernard Bou</a>
 */
public class FnPatternPointer extends Pointer
{
	/**
	 * Static field used to regenerate object, individually or as arrays
	 */
	static public final Parcelable.Creator<FnPatternPointer> CREATOR = new Parcelable.Creator<FnPatternPointer>()
	{
		@Override
		public FnPatternPointer createFromParcel(@NonNull final Parcel parcel)
		{
			return new FnPatternPointer(parcel);
		}

		@Override
		public FnPatternPointer[] newArray(final int size)
		{
			return new FnPatternPointer[size];
		}
	};

	/**
	 * Constructor from parcel, reads back fields IN THE ORDER they were written
	 */
	private FnPatternPointer(@NonNull final Parcel parcel)
	{
		super(parcel);
	}

	/**
	 * Constructor
	 *
	 * @param patternId pattern id
	 */
	public FnPatternPointer(final long patternId)
	{
		super(patternId);
	}
}

File: frameNet/src/main/java/org/sqlunet/framenet/FnPatternPointer.kt
/*
 * Copyright (c) 2023. Bernard Bou
 */
package org.sqlunet.framenet

import android.os.Parcel
import android.os.Parcelable
import org.sqlunet.Pointer

/**
 * Parcelable pattern
 *
 * @author [Bernard Bou](mailto:1313ou@gmail.com)
 */
class FnPatternPointer : Pointer {

    /**
     * Constructor from parcel, reads back fields IN THE ORDER they were written
     */
    private constructor(parcel: Parcel) : super(parcel)

    /**
     * Constructor
     *
     * @param patternId pattern id
     */
    constructor(patternId: Long) : super(patternId)

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<FnPatternPointer> {

        override fun createFromParcel(parcel: Parcel): FnPatternPointer {
            return FnPatternPointer(parcel)
        }

        override fun newArray(size: Int): Array<FnPatternPointer?> {
            return arrayOfNulls(size)
        }
    }
}
